src/main/java/com/google/sps/data/Keywords.java

```diff
@@ -0,0 +1,132 @@
+import com.google.cloud.language.v1.EncodingType;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * Provide methods for the storage, extraction, and retrieval of keywords.
+ */
+public final class Keywords {
+
+  private static final int MAX_NUM_KEYWORDS = 10;
+
+  /**
+   * @return a list of the 10 most salient keywords
+   */
+  public static List<String> getKeywords(String key) throws EntityNotFoundException {
+    try {
+      Key datastoreKey = KeyFactory.stringToKey(key);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Entity result = datastore.get(datastoreKey);
+    
+      Collection<String> keywordCollection = (Collection<String>) result.getProperty("keywords");
+      List<String> keywordList = new ArrayList<>();
+      int index = 0;
+      for (String keyword : keywordCollection) {
+        keywordList.add(keyword);
+    
+        index++;
+        if (index >= MAX_NUM_KEYWORDS) {
+          break;
+        }
+      }
+      return keywordList;
+    } catch (EntityNotFoundException ex) {
+      return new ArrayList<>();
+    }
+  }
+
+  /**
+   * Adds the labels from the analysis of an image into the datastore.
+   * @return the key pointing to the newly added datastore entity.
+   */
+  public static String addKeywords(List<EntityAnnotation> blobAnalysis) throws IOException {
+    String labels = "";
+    for (EntityAnnotation label : blobAnalysis) {
+      labels += label.getDescription() + ", ";
+    }
+    return addToDatastore(labels);
+  }
+
+  /**
+   * Adds the salient keywords from the analysis of a textual user input into the datastore.
+   * @return the key pointing to the newly added datastore entity.
+   */
+  public static String addKeywords(String message) throws IOException {
+    return addToDatastore(message);
+  }
+  
+  /**
+   * Adds salient keywords from the message into the datastore.
+   */
+  private static String addToDatastore(String message) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    AnalyzeEntitiesResponse entityResponse = analyzeEntity(message);
+    List<com.google.cloud.language.v1.Entity> entities = entityResponse.getEntitiesList();
```

Comment:
Maybe try to import this so you don't need to use the fully qualified name.

Reply:
There are two different Entity classes being used, and this is the only way to help Java differentiate between the two.